lib/OpenQA/Schema/Result/Assets.pm

```diff
@@ -133,22 +134,29 @@ sub delete {
 
 sub ensure_size {
     my ($self) = @_;
+    my $size = $self->size;
+    return $size if (defined($size));
+    return $self->refresh_size($size);
+}
+
+sub refresh_size {
+    my ($self, $current_size) = @_;
+    $current_size //= $self->size;
 
-    my $size = 0;
-    my @st   = stat($self->disk_file);
-    if (@st) {
+    my $new_size = 0;
+    my @stat = stat(my $disk_file = $self->disk_file);
```

Comment:
I really wonder if we could have a gru task doing this for us too, and still keep this around just in case the gru task still hasn't kicked in

Reply:
Forget the comment above, I see now that limit_assets is doing it